src/ophyd_async/core/_soft_signal_backend.py

```diff
@@ -50,10 +49,10 @@ def write_value(self, value: Any) -> Sequence[str]:
 
 
 @dataclass
-class SequenceEnumSoftConverter(SoftConverter[Sequence[EnumT]]):
-    datatype: type[EnumT]
+class SequenceEnumSoftConverter(SoftConverter[Sequence[EnumTypes]]):
+    datatype: type[EnumTypes]
 
-    def write_value(self, value: Any) -> Sequence[EnumT]:
+    def write_value(self, value: Any) -> Sequence[EnumTypes]:
```

Comment:
This was correct as it was, `Sequence[EnumT]` means "a sequence of instances of an enum type inferred from the datatype field, as long as the datatype is a subclass of EnumTypes", while `Sequence[EnumTypes]` means "sequence of anything that is an EnumType subclass" which is less strict

https://pyright-play.net/?code=GYJw9gtgBApgdgV2gSwgBzCALlAooiAKFEiiwE81k4BzKVDbKAFUpgDUBDEAGigGUYARwTwAxjEKExAG04BneXgIBBABT4kASgBchKAahoF8qbJPKkAIQ0Fd%2Bw8cVTNEVmhhKAvJYgqoAD6%2BVoSuzFA%2B7hzcagBEYbF8AEZgCHAAJhG%2BUfJaUukwwFAQ5AD6wGliapw6AsKicBIA2mEAulpQALQAfNl6hlAgMFgIIHBQnE0ADK35hcVlFY0ATNW1giLiMC0EOe1dvWFs8v2GQyNjE9OzZnKKLAAW1DS21vYD-j6x8lggz7FSAAeWRK5UqaiazCetAAdCp2oQhgA3GCcGSlCgeNSAvKEcggxaVVaQ6E0OEI5Go9GYmBqch5IA